### PR TITLE
Adding documentation for server configuration

### PR DESCRIPTION
**Pull request by desiam**
### Context

Adding documentation for common server configuration properties.

### Testing
Preview: [server_configuration.pdf](https://github.com/Yelp/nrtsearch/files/9023616/server_configuration.pdf)

### CODE REVIEW

File: docs/server_configuration.rst [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+     - true
+
+   * - restoreState
+     - bool
+     - Enables loading state from external storage on startup
+     - false
+
+   * - restoreFromIncArchiver
+     - bool
+     - If enabled, uses the incremental archiver when restoring state
```
Comment (aprudhomme) [inline docs/server_configuration.rst:101]: This would apply to index data as well as state.

File: docs/server_configuration.rst [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+     - false
+
+   * - backupWithIncArchiver
+     - bool
+     - If enabled, uses the incremental archiver for backups
+     - false
+
+   * - deadlineCancellation
+     - bool
+     - Enables gRPC deadline based cancellation of requests
```
Comment (aprudhomme) [inline docs/server_configuration.rst:111]: This cancels requests early if they hit their deadline. Only supported by the search endpoint at the moment.

File: docs/server_configuration.rst [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+     - false
+
+   * - plugins
+     - list
+     - List of plugins located in the ``pluginSearchPath`` to load
+     - []
+
+   * - pluginSearchPath
+     - str
+     - Search path for plugins. The server will try to find the first directory in the search path matching a given plugin.
```
Comment (aprudhomme) [inline docs/server_configuration.rst:121]: Paths within this string are separated by the system path separator character (i.e. `:` for linux, `;` for windows)

File: docs/server_configuration.rst [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+   :header-rows: 1
+
+   * - Property
+     - Type
+     - Description
+     - Default
+
+   * - maxWarmingQueries
+     - int
+     - Maximum number of queries to send during warming
```
Comment (aprudhomme) [inline docs/server_configuration.rst:169]: I believe this is the maximum queries to store for warming. Replay will process all the queries that were backed up.

**Review by aprudhomme — COMMENTED** [review]
There seems to be a lot of config options not present here.

**Comment by desiam** [comment]
Updated PDF of rendered [server_configuration.pdf](https://github.com/Yelp/nrtsearch/files/9031598/server_configuration.pdf)

**Comment by desiam** [comment]
> There seems to be a lot of config options not present here.

Yeah, there's quite a bit of configuration. Will continue to add properties to this PR.

**Review by sarthakn7 — APPROVED** [review]

**Pull request merged**